menas/ui/components/dataset/conformanceRule/ConformanceRuleDialog.js

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class ConformanceRuleDialog {
+
+  constructor(controller) {
+    this._model = sap.ui.getCore().getModel();
+    this.model.setProperty("/rules", this.rules);
+    this.model.setProperty("/dataTypes", this.dataTypes);
+
+    this._formFragments = {};
+
+    const eventBus = sap.ui.getCore().getEventBus();
+    this._datasetService = new DatasetService(this.model, eventBus);
+    this._mappingTableService = new MappingTableService(this.model, eventBus);
+    this._controller = controller;
+    const dialogFactory = new JoinConditionDialogFactory(this.controller, sap.ui.core.Fragment.load);
+    this._addJoinConditionDialog = dialogFactory.getDialog();
+  }
+
+  get controller() {
+    return this._controller;
+  }
+
+  get model() {
+    return this._model;
+  }
+
+  get formFragments() {
+    return this._formFragments;
+  }
+
+  get datasetService() {
+    return this._datasetService;
+  }
+
+  get mappingTableService() {
+    return this._mappingTableService;
+  }
+
+  get addJoinConditionDialog() {
+    return this._addJoinConditionDialog;
+  }
+
+  get rules() {
+    return [
+      {
+        _t: "CastingConformanceRule",
+        schemaFieldSelectorSupportedRule: true
+      },
+      {
+        _t: "ConcatenationConformanceRule",
+        schemaFieldSelectorSupportedRule: false
+      },
+      {
+        _t: "DropConformanceRule",
+        schemaFieldSelectorSupportedRule: false
+      },
+      {
+        _t: "LiteralConformanceRule",
+        schemaFieldSelectorSupportedRule: false
+      },
+      {
+        _t: "MappingConformanceRule",
+        schemaFieldSelectorSupportedRule: true
+      },
+      {
+        _t: "NegationConformanceRule",
+        schemaFieldSelectorSupportedRule: true
+
+      },
+      {
+        _t: "SingleColumnConformanceRule",
+        schemaFieldSelectorSupportedRule: true
+      },
+      {
+        _t: "SparkSessionConfConformanceRule",
+        schemaFieldSelectorSupportedRule: false
+      },
+      {
+        _t: "UppercaseConformanceRule",
+        schemaFieldSelectorSupportedRule: true
+      }
+    ]
+  }
+
+  get dataTypes() {
+    return [
+      {type: "boolean"},
+      {type: "byte"},
+      {type: "short"},
+      {type: "integer"},
+      {type: "long"},
+      {type: "float"},
+      {type: "double"},
+      {type: "decimal(38,18)"},
```

Comment:
Should the precision and scale be hardcoded here?

Reply:
It's a remnant from the old Menas. Of course it would be better to make these editable, but that's for a different task.